### PR TITLE
Shift horse's data encoding to back-end.

### PR DESCRIPTION
**Pull request by aguxez**
Initially we were encoding horse's data on the front-end, leaving us open for a man in the middle attacks. We're encoding horse's data on the back-end and sending it through the socket now.

### CODE REVIEW

File: lib/cryptofield_api/horse_workers/horse_state.ex [hunk 1]
```diff
@@ -236,6 +241,18 @@ defmodule CryptofieldApi.HorseState do
     }
   end
 
+  defp generate_horse_hash(data, tries \\ 0)
+  defp generate_horse_hash(_data, 5), do: ""
```
Comment (girishramnani) [inline lib/cryptofield_api/horse_workers/horse_state.ex]: shouldn't we returns an error tuple `{:error, :trial_limit}` or something?

Comment (aguxez) [reply lib/cryptofield_api/horse_workers/horse_state.ex]: Yes, we should. I just wanted to keep it simple with the return values here. We need to change the whole process call to return the error. I'll change it :)

Comment (girishramnani) [reply lib/cryptofield_api/horse_workers/horse_state.ex]: awesome

Comment (girishramnani) [reply lib/cryptofield_api/horse_workers/horse_state.ex]: its upto you, we can solve it later

Comment (aguxez) [reply lib/cryptofield_api/horse_workers/horse_state.ex]: Let's leave it as is for now then. We'll push a change throughout the week.

**Pull request merged**